### ISSUE TITLE
Error while updating

### ISSUE THREAD

**Issue by tiagaoalb** [LunarVim/LunarVim#2517]
### Problem description

Got the error `[ERROR] lvim: "Update failed! Please pull the changes manually instead." file="git.lua", line=72%` while updating lvim.

### LunarVim version

latest

### Neovim version (>= 0.6.1)

0.6.1

### Operating system/version

macOS 12.13.1

### Relevant log output

```shell
null-ls: require("null-ls.health").check()
========================================================================
  - INFO: no sources registered

nvim: health#nvim#check
========================================================================
## Configuration
  - WARNING: Missing user config file: /Users/tiagaoalb/.config/nvim/init.vim
    - ADVICE:
      - :help |init.vim|

## Performance
  - OK: Build type: Release

## Remote Plugins
  - OK: Up to date

## terminal
  - INFO: key_backspace (kbs) terminfo entry: key_backspace=\177
  - INFO: key_dc (kdch1) terminfo entry: key_dc=\E[3~
  - INFO: $COLORTERM='truecolor'

nvim-lsp-installer: require("nvim-lsp-installer.health").check()
========================================================================
## nvim-lsp-installer report
  - OK: neovim version >= 0.6.0
  - WARNING: **Go**: not available
  - WARNING: **Composer**: not available
  - WARNING: **PHP**: not available
  - WARNING: **julia**: not available
  - ERROR: **wget**: not available
  - OK: **bash**: `GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin21)`
  - OK: **tar**: `bsdtar 3.5.1 - libarchive 3.5.1 zlib/1.2.11 liblzma/5.0.5 bz2lib/1.0.8 `
  - OK: **gzip**: `Apple gzip 353.100.22`
  - OK: **curl**: `curl 7.79.1 (x86_64-apple-darwin21.0) libcurl/7.79.1 (SecureTransport) LibreSSL/3.3.5 zlib/1.2.11 nghttp2/1.45.1`
  - WARNING: **java**: not available
  - WARNING: **javac**: not available
  - OK: **python3**: `Python 3.10.4`
  - OK: **node**: `v16.13.0`
  - OK: **Ruby**: `ruby 2.6.8p205 (2021-07-07 revision 67951) [universal.x86_64-darwin21]`
  - OK: **sh**: `Ok`
  - OK: **RubyGem**: `3.0.3.1`
  - OK: **npm**: `8.3.0`
  - OK: **pip3**: `pip 22.0.4 from /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pip (python 3.10)`

nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - OK: `tree-sitter` found 0.20.6 (ccd6bf554d922596ce905730d98a77af368bba5c) (parser generator, only needed for :TSInstallFromGrammar)
  - OK: `node` found v16.13.0 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
    Version: Apple clang version 13.1.6 (clang-1316.0.21.2.3)
  - OK: Neovim was compiled with tree-sitter runtime ABI version 13 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - json           ‚úì ‚úì ‚úì ‚úì . 
  - python         ‚úì ‚úì ‚úì ‚úì ‚úì 
  - rust           ‚úì ‚úì ‚úì ‚úì ‚úì 
  - javascript     ‚úì ‚úì ‚úì ‚úì ‚úì 
  - tsx            ‚úì ‚úì ‚úì ‚úì ‚úì 
  - typescript     ‚úì ‚úì ‚úì ‚úì ‚úì 
  - yaml           ‚úì ‚úì ‚úì ‚úì ‚úì 
  - lua            ‚úì ‚úì ‚úì ‚úì ‚úì 
  - css            ‚úì . ‚úì ‚úì ‚úì 
  - java           ‚úì ‚úì . ‚úì ‚úì 
  - c              ‚úì ‚úì ‚úì ‚úì ‚úì 
  - bash           ‚úì ‚úì ‚úì . ‚úì 

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}

provider: health#provider#check
========================================================================
## Clipboard (optional)
  - OK: Clipboard tool found: pbcopy

## Python 2 provider (optional)
  - INFO: pyenv: Path: /usr/local/Cellar/pyenv/2.2.5/libexec/pyenv
  - INFO: pyenv: Root: /Users/tiagaoalb/.pyenv
  - WARNING: No Python executable found that can `import neovim`. Using the first available executable for diagnostics.
  - ERROR: Python provider error:
    - ADVICE:
      - provider/pythonx: Could not load Python 2:
          /Library/Frameworks/Python.framework/Versions/2.7/bin/python2 does not have the "neovim" module. :help |provider-python|
          /Library/Frameworks/Python.framework/Versions/2.7/bin/python2.7 does not have the "neovim" module. :help |provider-python|
          python2.6 not found in search path or not executable.
          /Library/Frameworks/Python.framework/Versions/2.7/bin/python does not have the "neovim" module. :help |provider-python|
  - INFO: Executable: Not found

## Python 3 provider (optional)
  - INFO: pyenv: Path: /usr/local/Cellar/pyenv/2.2.5/libexec/pyenv
  - INFO: pyenv: Root: /Users/tiagaoalb/.pyenv
  - INFO: `g:python3_host_prog` is not set.  Searching for python3 in the environment.
  - INFO: Executable: /Users/tiagaoalb/.pyenv/versions/3.9.6/bin/python3
  - ERROR: Command error (job=25, exit code 1): `'/Users/tiagaoalb/.pyenv/versions/3.9.6/bin/python3' -c 'import sys; sys.path = list(filter(lambda x: x != "", sys.path)); import neovim; print(neovim.__file__)'` (in '/Users/tiagaoalb')
    stderr: Traceback (most recent call last):  File "<string>", line 1, in <module>ModuleNotFoundError: No module named 'neovim'
  - INFO: Python version: 3.9.6
  - INFO: pynvim version: unable to load neovim Python module
  - ERROR: pynvim is not installed.
    Error: unable to load neovim Python module
    - ADVICE:
      - Run in shell: /Users/tiagaoalb/.pyenv/versions/3.9.6/bin/python3 -m pip install pynvim

## Python virtualenv
  - OK: no $VIRTUAL_ENV

## Ruby provider (optional)
  - INFO: Ruby: ruby 2.6.8p205 (2021-07-07 revision 67951) [universal.x86_64-darwin21]
  - WARNING: `neovim-ruby-host` not found.
    - ADVICE:
      - Run `gem install neovim` to ensure the neovim RubyGem is installed.
      - Run `gem environment` to ensure the gem bin directory is in $PATH.
      - If you are using rvm/rbenv/chruby, try "rehashing".
      - See :help |g:ruby_host_prog| for non-standard gem installations.

## Node.js provider (optional)
  - INFO: Node.js: v16.13.0
  - INFO: Nvim node.js host: /Users/tiagaoalb/.config/yarn/global//node_modules/neovim/bin/cli.js
  - OK: Latest "neovim" npm/yarn package is installed: 4.10.1

## Perl provider (optional)
  - ERROR: perl provider error:
    - ADVICE:
      - "Neovim::Ext" cpan module is not installed

telescope: require("telescope.health").check()
========================================================================
## Checking for required plugins
  - OK: plenary installed.
  - OK: nvim-treesitter installed.

## Checking external dependencies
  - OK: rg: found ripgrep 13.0.0
  - OK: fd: found fd 8.3.2

## ===== Installed extensions =====

## Telescope Extension: `fzf`
  - INFO: No healthcheck provided

## Telescope Extension: `projects`
  - INFO: No healthcheck provided

vim.lsp: require("vim.lsp.health").check()
========================================================================
  - INFO: LSP log level : WARN
  - INFO: Log path: /Users/tiagaoalb/.cache/nvim/lsp.log
  - INFO: Log size: 1 KB

vim.treesitter: require("vim.treesitter.health").check()
========================================================================
  - INFO: Runtime ABI version : 13
  - OK: Loaded parser for bash: ABI version 13
  - OK: Loaded parser for c: ABI version 13
  - OK: Loaded parser for css: ABI version 13
  - OK: Loaded parser for java: ABI version 13
  - OK: Loaded parser for javascript: ABI version 13
  - OK: Loaded parser for json: ABI version 13
  - OK: Loaded parser for lua: ABI version 13
  - OK: Loaded parser for python: ABI version 13
  - OK: Loaded parser for rust: ABI version 13
  - OK: Loaded parser for tsx: ABI version 13
  - OK: Loaded parser for typescript: ABI version 13
  - OK: Loaded parser for yaml: ABI version 13

which_key: health#which_key#check
========================================================================
## WhichKey: checking conflicting keymaps
  - WARNING: buffer-local keymap overriding global for mode **"n"**, buf: 0, lhs: **"<leader>sr"**
  - INFO: old rhs: `<CMD>Telescope oldfiles<CR>`
  - INFO: new rhs: `<cmd>Telescope oldfiles<cr>`
  - WARNING: buffer-local keymap overriding global for mode **"n"**, buf: 0, lhs: **"<leader>st"**
  - INFO: old rhs: `<CMD>Telescope live_grep<CR>`
  - INFO: new rhs: `<cmd>Telescope live_grep<cr>`
  - WARNING: buffer-local keymap overriding global for mode **"n"**, buf: 0, lhs: **"<leader>f"**
  - INFO: old rhs: `<CMD>Telescope find_files<CR>`
  - INFO: new rhs: `<cmd>lua require("which-key").execute(1)<cr>`
  - WARNING: buffer-local keymap overriding global for mode **"n"**, buf: 0, lhs: **"<leader>Lc"**
  - INFO: old rhs: `<CMD>edit /Users/tiagaoalb/.config/lvim/config.lua <CR>`
  - INFO: new rhs: `<cmd>edit /Users/tiagaoalb/.config/lvim/config.lua<cr>`

```

### Screenshots

![Screen Shot 2022-04-25 at 10 49 03](https://user-images.githubusercontent.com/26368452/165102575-956cd487-67fa-4956-9029-aee8f9c03820.png)

**Comment by abzcoding** [comment]
<img width="622" alt="Screen Shot 2022-04-25 at 6 37 39 PM" src="https://user-images.githubusercontent.com/10992695/165105979-1a2b5d44-82be-4f91-a8c8-c495fa6bd348.png">


it seems like you've manually changed some of the lunarvim files inside `~/.local/share/lunarvim/lvim` ( which you should not touch 😅 )

so you have to manually fix it
```shell
cd ~/.local/share/lunarvim/lvim
git status -- to see what are the changes
git reset --hard -- to revert the changes
git pull -- to pull the latest changes
```

**Comment by tiagaoalb** [comment]
> 

Man, I dont think so 😄 .
Just did what you asked and broke lvim.

**Comment by tiagaoalb** [comment]
Got this messages after pull.
![Screen Shot 2022-04-25 at 14 16 36](https://user-images.githubusercontent.com/26368452/165140152-ba43d9d5-0b24-4a56-9888-277f96c08f4e.png)

**Comment by abzcoding** [comment]
> Got this messages after pull.
> ![Screen Shot 2022-04-25 at 14 16 36](https://user-images.githubusercontent.com/26368452/165140152-ba43d9d5-0b24-4a56-9888-277f96c08f4e.png)

ok run a `:LvimCacheReset` and `:PackerSync` these issues are caused by an older version of plugins

**Comment by abzcoding** [comment]
if it is getting too much annoying, just uninstall/reinstall it

**Comment by tiagaoalb** [comment]
Uninstall and install solved for me. Thank you for your time 🤘🏻

**Comment by scottrbaxter** [comment]
I realize this is closed, but I also just experienced this issue (stable, not rolling). Went through all of the steps to no avail. Renamed `~/.local/share/lunarvim` and re-installed following the recommended process:

```
bash <(curl -s https://raw.githubusercontent.com/lunarvim/lunarvim/master/utils/installer/install.sh)
```

I no longer have this issue. Don't believe I touched anything in this directory, so unsure why this issue happened... but I could look through the archived version if it'd help troubleshoot in the future. Fortunately removing and reinstalling fixes this issue 😊
